### DISCUSSION TITLE
Feature idea: BYO virtual display for hiding windows

### DISCUSSION THREAD

**Discussion by jaybonthius** [nikitabobko/AeroSpace#623] [feature-ideas]
Aerospace currently uses the bottom corners of each display to hide windows. As the documentation [points out](https://nikitabobko.github.io/AeroSpace/guide#proper-monitor-arrangement), this method imposes limitations on monitor configurations. Many popular setups, such as [triple-screen](https://www.google.com/search?tbm=isch&q=triple+monitor+setup) and [tie-fighter](https://www.google.com/search?tbm=isch&q=tie+fighter+monitor+setup) arrangements, are incompatible with Aerospace.

It's possible to create virtual displays that exist only in memory. In fact, the popular application [BetterDisplay](https://github.com/waydabber/BetterDisplay) recently added a feature to manage virtual displays.

I propose an Aerospace feature that would allow users to specify a virtual display to use as a "free space" for hiding windows. This would accommodate any monitor configuration the user desires. To keep Aerospace as lightweight as possible, this could be implemented as bring-your-own-virtual-display—users could then use whichever virtual display management app they prefer and simply tell Aerospace which display to use for hiding.

## Checklist

- [X] I've searched for duplicates. My search queries were (please don't forget to search in closed issues too):
  - virtual display
  - virtual
  - betterdisplay
- [X] I've read the guide https://nikitabobko.github.io/AeroSpace/guide
- [X] I've searched in the documentation (https://nikitabobko.github.io/AeroSpace/guide & https://nikitabobko.github.io/AeroSpace/commands). My search queries were:
  - virtual display
  - virtual

**Comment by nikitabobko** [comment]
I had this idea in my mind. It won't work because:
1. If monitors have different resolutions, it will be a performance disaster #149 
2. Some apps may not like to be often moved between monitors (like chrome: #289)
3. If AeroSpace crashes, users will loose windows on the invisible monitor
4. I don't like providing config options for choosing between different trade-offs
5. To my taste, a "virtual monitor" hack is just intuitively too much invasive to be reliable, there might be other problems

**Reply by froggologies** [reply]
How about hiding it to the dock?
